infrastructure/devicepathresolver/scsi_id_device_path_resolver.go

```diff
@@ -0,0 +1,108 @@
+package devicepathresolver
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+	boshlog "github.com/cloudfoundry/bosh-utils/logger"
+	boshsys "github.com/cloudfoundry/bosh-utils/system"
+)
+
+// Resolves device path by performing a SCSI rescan then looking under
+// "/dev/disk/by-id/*uuid" where "uuid" is the cloud ID of the disk
+type scsiIDDevicePathResolver struct {
+	diskWaitTimeout time.Duration
+	fs              boshsys.FileSystem
+	logger          boshlog.Logger
+}
+
+func NewScsiIDDevicePathResolver(
+	diskWaitTimeout time.Duration,
+	fs boshsys.FileSystem,
+	logger boshlog.Logger,
+) DevicePathResolver {
+	return scsiIDDevicePathResolver{
+		diskWaitTimeout: diskWaitTimeout,
+		fs:              fs,
+		logger:          logger,
+	}
+}
+
+const logTag = "scsiIDresolver"
```

Comment:
let's put this tag into the struct e.g. https://github.com/cloudfoundry/bosh-agent/blob/master/agent/script/parallel_script.go#L28